### PR TITLE
GA Migration Java PostCommit Nexmark Direct, Flink, Spark

### PR DESCRIPTION
**Pull request by andoni-guzman**
This PR is part of the migration from jenkins to github actions.

This PR contains this tests

Java PostCommit Nexmark Direct
Java PostCommit Nexmark Flink
Java PostCommit Nexmark Spark

3 Files were added to the workflows folder of github actions
- job-postcommit-java-nexmark-direct and
- job-postcommit-java-nexmark-flink 
- job-postcommit-java-nexmark-spark 
3 Files were deleted, this files were the groovy files for jenkins
- Nexmark Direct
- Nexmark Flink
- Nexmark Spark
-
Jenkins readme was modified deleting migrated tests.
CI.md was modified adding the new GA workflows

Waiting for this PR before merge https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by fernando-wizeline** [comment]
LGTM!

File: .github/workflows/job-postcommit-java-nexmark-direct.yml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+    timeout-minutes: 240
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false        
+      - name: Run *** RUN NEXMARK IN BATCH MODE USING DIRECT RUNNER ***
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-nexmark-direct.yml:54]:
```suggestion
      - name: Run Nexmark in Batch mode on the direct runner
```

Comment (damccorm) [reply .github/workflows/job-postcommit-java-nexmark-direct.yml:54]: Same comment applies elsewhere - please remove bold/all caps

File: .github/workflows/job-postcommit-java-nexmark-direct.yml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+permissions: read-all
+
+env:
+  BIGQUERYTABLE: nexmark
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-nexmark-direct.yml:30]: These environment variables are used in a number of workflows. Should we refactor them to our shared variable config?

File: .github/workflows/job-postcommit-java-nexmark-direct.yml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+              --baseInfluxMeasurement=$BASEINFLUXMEASUREMENT 
+              --exportSummaryToInfluxDB=$EXPORTSUMMARYTOINFLUXDB 
+              --influxRetentionPolicy=$INFLUXRETENTIONPOLICY 
+              --queryLanguage=sql 
+              --streaming=true 
+              --suite=SMOKE 
+              --manageResources=false 
+              --monitorJobs=true 
+              --enforceEncodability=true 
+              --enforceImmutability=true "
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-nexmark-direct.yml:201]: Is the only difference here the streaming variable and the queryLanguage? If so, can we just make these separate jobs and use a matrix to run with each variable type?

Comment (damccorm) [reply .github/workflows/job-postcommit-java-nexmark-direct.yml:201]: Same comment applies below in the other workflows. In fact, if the runner is the only difference, we could probably just consolidate to a single workflow and use matrix builds on the runner.

Comment (andoni-guzman) [reply .github/workflows/job-postcommit-java-nexmark-direct.yml:201]: Working in the changes.

**Comment by andoni-guzman** [comment]
Hi @damccorm 
I made some changes in this PR
- I use the test-properties file to get all the variables in all the tests of this PR.
- All the tests are now using a matrix, 
- In the case of the tests related with spark,  I use 2 jobs, the reason that are in different jobs in the workflow and not in a matrix are that the runner are different and one runner needs a special flag to work, using this in a matrix cause the failures of the pipeline, for example using the runner SparkRunner  with the configuration of SparkStructuredStreamingRunner 
If some changes are needed please let me know

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**